website/server/controllers/api-v3/groups.js

```diff
@@ -402,6 +409,11 @@ api.leaveGroup = {
     }
 
     await group.leave(user, req.query.keep);
+
+    if (user.newMessages[group._id]) {
```

Comment:
To me, the only point of the if check is to save a call to markModified if it isn't needed. You could arguably do the logic without the check, and it would make no difference except the call to markModified. 

So let's put the if check in the function.